osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/misc/GetPowerQualityProfileCommandExecutor.java

```diff
@@ -463,43 +465,107 @@ private ProfileEntryValueDto makeNumericProfileEntryValueDto(final DataObject da
-                    final DataObject scalerUnitDataObject = scalerUnitResult.get(0).getResultData();
-                    result.add(new ScalerUnitInfo(logicalName, classId, scalerUnitDataObject));
-                } else {
-                    result.add(new ScalerUnitInfo(logicalName, classId, null));
+
+                if (logicalNames.contains(logicalName)) {
+                    selectableCaptureObjects.put(positionInDataObjectsList,
+                            new CaptureObjectDefinitionDto(cosemObjectDefinitionDto.getClassId(),
+                                    new ObisCodeValuesDto(logicalName),
+                                    (byte) cosemObjectDefinitionDto.getAttributeIndex(),
+                                    cosemObjectDefinitionDto.getDataIndex()));
                 }
+                positionInDataObjectsList++;
             }
         }
 
-        return result;
+        return selectableCaptureObjects;
+    }
+
+    private List<ScalarUnitInfo> createScalarUnitInfos(final DlmsConnectionManager conn, final DlmsDevice device,
+            final List<GetResult> captureObjects) throws ProtocolAdapterException {
+
+        List<ScalarUnitInfo> scalarUnitInfos = new ArrayList<>();
+
+        // there is always only one GetResult
+        for (final GetResult captureObjectResult : captureObjects) {
+
+            final List<DataObject> dataObjects = captureObjectResult.getResultData().getValue();
+
+            for (DataObject dataObject : dataObjects) {
+                scalarUnitInfos.add(createScalarUnitInfo(conn, device, dataObject));
+            }
+        }
+
+        return scalarUnitInfos;
+    }
+
+    private ScalarUnitInfo createScalarUnitInfo(final DlmsConnectionManager conn, final DlmsDevice device,
+            final DataObject dataObject) throws ProtocolAdapterException {
+
+        final CosemObjectDefinitionDto cosemObjectDefinitionDto = this.dlmsHelper
+                .readObjectDefinition(dataObject, CAPTURE_OBJECT);
+
+        final int classId = cosemObjectDefinitionDto.getClassId();
+        final String logicalName = cosemObjectDefinitionDto.getLogicalName().toString();
+
+        if (this.hasScalarUnit(classId)) {
+            final AttributeAddress addr = new AttributeAddress(classId, logicalName, SCALAR_UNITS_MAP.get(classId));
+            final List<GetResult> scalarUnitResult = this.dlmsHelper
+                    .getAndCheck(conn, device, "retrieve scalar unit for capture object", addr);
+            final DataObject scalarUnitDataObject = scalarUnitResult.get(0).getResultData();
+            return new ScalarUnitInfo(logicalName, classId, scalarUnitDataObject);
+        } else {
+            return new ScalarUnitInfo(logicalName, classId, null);
+        }
+    }
+
+    private static List<String> getLogicalNamesPublicDefinableLoadProfile() {
+
+        return Arrays
+                .asList("0.0.1.0.0.255", "1.0.32.32.0.255", "1.0.52.32.0.255", "1.0.72.32.0.255", "1.0.32.36.0.255",
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "0.0.1.0.0.255" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1192)